pkg/loader/fileloader.go

```diff
@@ -17,32 +17,75 @@ limitations under the License.
 package loader
 
 import (
+	"fmt"
+	"os"
 	"path/filepath"
 
 	"k8s.io/kubectl/pkg/kinflate/util/fs"
 )
 
-// Implements internal interface schemeLoader.
+const currentDir = "."
+
+// Internal implementation of SchemeLoader interface.
 type fileLoader struct {
 	fs fs.FileSystem
 }
 
-func newFileLoader(fs fs.FileSystem) (schemeLoader, error) {
-	return &fileLoader{fs: fs}, nil
+// NewFileLoader returns a SchemeLoader to handle a file system.
+func NewFileLoader(fs fs.FileSystem) SchemeLoader {
+	return &fileLoader{fs: fs}
+}
+
+// Is the location calculated with the root and location params a full file path.
+func (l *fileLoader) IsScheme(root string, location string) bool {
+	fullFilePath, err := l.FullLocation(root, location)
+	if err != nil {
+		return false
+	}
+	return filepath.IsAbs(fullFilePath)
+}
+
+// Returns the directory of the calculated full file path.
+// Example: "/home/seans/project", "subdir/file.txt" -> "/home/seans/project/subdir".
+func (l *fileLoader) Root(root string, location string) (string, error) {
+	fullFilePath, err := l.FullLocation(root, location)
+	if err != nil {
+		return "", err
+	}
+	return filepath.Dir(fullFilePath), nil
 }
 
-// Join the root path with the location path.
-func (l *fileLoader) fullLocation(root string, location string) string {
+// If location is a full file path, then ignore root. If location is relative, then
```

Comment:
This documentation piece is not printed, because it's documenting a private struct. It's too bad because the exposed documentation is not very clear.